### DISCUSSION TITLE
Migration Postgres to Snowflake ?

### DISCUSSION THREAD

**Discussion by LouisMonierSpark** [elevate/elevate.metabase.tools#28] [Q&A]
Hey folks, 
We want to migrate from a Postgres database to a Snowflake database in Metabase. As you know Snowflake stores database identifiers (schema names, table names, column names) in uppercase natively. It doesn't mean we can't query a table with lowercase, but Metabase fetches those identifiers in uppercase when connected. 
So, what happens in our case here, is we have the exact same structure in our Postgres database and our new Snowflake database. The only difference in Metabase is the identifiers are in lowercase for the Postgres and in uppercase for Snowflake in Metabase. This small difference makes the migration a lot harder as it breaks the field filters of hundreds of saved native queries.

Hence, my question: does this repo handles the migration of native Metabase questions written initially using a Postgres database to a Snowflake database ? To be more specific, I've successively retrieved the IDs of all the columns from the Postgres and Snowflake to create a matching and make sure the field filters do not break. However, the hardest part is to transfer the visualization settings. Switching to Snowflake also breaks the visualization of the question.

Other option would be to use quotes when creating the Snowflake elements: "schema_name"."table_name". The quotes enables to store database identifiers in lowercase in Snowflake. But it's not possible to query them without quotes in the Metabase saved questions. So, it's not an option. 

What would you recommend to migrate seamlessly ? 

Thank you !

**Comment by mausch** [comment]
Hi, I've only used Metabase with Postgres, and I've never used Snowflake in any context.
This tool only dumps the current state of Metabase to a JSON file. If you know what needs to be translated and how, maybe you could write a small program to traverse the JSON file and convert what needs to be converted?
